native/acter/src/api/timeline/item.rs

```diff
@@ -127,123 +130,158 @@ pub struct TimelineEventItem {
     edited: bool,
 }
 
-impl TimelineEventItem {
-    pub(crate) fn new(event: &EventTimelineItem, my_id: OwnedUserId) -> Self {
-        let mut me = TimelineEventItemBuilder::default();
-
-        me.event_id(event.event_id().map(ToOwned::to_owned))
-            .txn_id(event.transaction_id().map(ToOwned::to_owned))
-            .sender(event.sender().to_owned())
-            .send_state(
-                event
-                    .send_state()
-                    .map(|s| EventSendState::new(s, event.local_echo_send_handle())),
-            )
-            .origin_server_ts(event.timestamp().get().into())
-            .read_receipts(
-                event
-                    .read_receipts()
-                    .iter()
-                    .map(|(u, receipt)| (u.to_string(), receipt.clone()))
-                    .collect(),
-            )
-            .reactions(
-                event
-                    .content()
-                    .reactions()
-                    .iter()
-                    .map(|(key, group)| {
-                        (
-                            key.clone(),
-                            group
-                                .iter()
-                                .map(|(sender_id, info)| {
-                                    ReactionRecord::new(
-                                        sender_id.clone(),
-                                        info.timestamp,
-                                        *sender_id == my_id,
-                                    )
-                                })
-                                .collect::<Vec<ReactionRecord>>(),
-                        )
-                    })
-                    .collect(),
-            )
-            .editable(event.is_editable()); // which means _images_ can't be edited right now ... but that is probably fine
-
-        match event.content() {
+impl TimelineEventItemBuilder {
+    fn parse_content(
+        self: &mut TimelineEventItemBuilder,
+        content: &SdkTimelineItemContent,
+        when: u64,
+        my_id: OwnedUserId,
+    ) {
+        self.origin_server_ts(when);
+        match content {
```

Comment:
mostly a move to make it easier to reuse for both the original and replied-to content